engine/src/main/java/com/sandboni/core/engine/BuilderPattern.java

```diff
@@ -1,10 +1,12 @@
 package com.sandboni.core.engine;
 
+import com.sandboni.core.scm.exception.SourceControlException;
+
 import java.util.function.Consumer;
 
 public interface BuilderPattern <R, T extends BuilderPattern>{
 
     T with(Consumer<T> builderFunction);
 
-    R build();
+    R build() throws SourceControlException;
```

Comment:
I don't see a reason to throw an Exception in the build() method. If there's a problem while building the object, then it should be a RuntimeException. Also BuilderPattern is a generic interface and SourceControlException is specific to scm context. Just leave it as R build();

Reply:
Agreed, this was due to a bad pull. Will update properly, I appreciate the catch + feedback.